servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/PublisherGroupToManyTest.java

```diff
@@ -109,4 +112,60 @@ void groupToMany(boolean onError) throws InterruptedException {
             group3Sub.awaitOnComplete();
         }
     }
+
+    @ParameterizedTest(name = "{displayName} [{index}]: applyMulticast={0}")
+    @ValueSource(booleans = {true, false})
+    void groupedPublisherResubscribe(boolean applyMulticast) throws InterruptedException {
+        PublisherSource.Subscription groupSubscription = groupSub.awaitSubscription();
+        groupSubscription.request(Long.MAX_VALUE);
+        subscription.awaitRequestN(2);
+        source.onNext(1);
+        GroupedPublisher<GroupColor, Integer> group1 = groupSub.takeOnNext();
+        assertThat(group1, notNullValue());
+        Publisher<Integer> group1Pub = group1;
+        if (applyMulticast) {
+            group1Pub = group1Pub.multicast(1, false);
+        }
+        assertThat(group1.key(), is(GroupColor.GREEN));
+        toSource(group1Pub).subscribe(group1Sub);
+        GroupedPublisher<GroupColor, Integer> group2 = groupSub.takeOnNext();
+        assertThat(group2, notNullValue());
+        assertThat(group2.key(), is(GroupColor.BLUE));
+        toSource(group2).subscribe(group2Sub);
+
+        PublisherSource.Subscription subscription1 = group1Sub.awaitSubscription();
+        PublisherSource.Subscription subscription2 = group2Sub.awaitSubscription();
+
+        subscription1.request(1);
+        assertThat(group1Sub.takeOnNext(), is(1));
+        subscription2.request(1);
+        assertThat(group2Sub.takeOnNext(), is(1));
+
+        subscription1.cancel();
+        TestPublisherSubscriber<Integer> group1SubNew = new TestPublisherSubscriber<>();
+        toSource(group1Pub).subscribe(group1SubNew);
+
+        source.onNext(3);
+        if (applyMulticast) {
+            PublisherSource.Subscription subscription1New = group1SubNew.awaitSubscription();
+            subscription1New.request(1);
+            assertThat(group1SubNew.takeOnNext(), is(3));
+        } else {
+            assertThat(group1SubNew.awaitOnError(), is(instanceOf(DuplicateSubscribeException.class)));
```

Comment:
It was hard to implement a full reproducer for the partitioned client, so I decided to show the problem in `PublisherGroupToManyTest`